_pytest/capture.py

```diff
@@ -252,6 +252,8 @@ def writelines(self, linelist):
         self.write(data)
 
     def __getattr__(self, name):
+        if name == 'name':
+            return repr(self.buffer)
         return getattr(object.__getattribute__(self, "buffer"), name)
```

Comment:
This code was already there, I think the idea is that `EncodedFile` should act as a proxy to the underlying `buffer`.